### ISSUE TITLE
[A11ySev2] [Keyboard Navigation -Dashboards with Grafana - New Dashboard]: Keyboard focus moves to background elements after selecting "Select a Resource" button under the Queries tab.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118778]
## Steps to reproduce

Repro Steps: 
  
 
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.  
3. Select already existing data as 'Azure Insights Key Vaults' under All Dashboards tab.  
4. Navigate to toolbar and select edit button. Now select Add Visualization from Add menu. 
5. Navigate Queries tab and Select 'Select a Resource' button. 
6. Navigate all controls under 'Select a Resource' dialog. 
7. Observe that the keyboard focus shifts to the background elements after opening the select a Resource dialog. 

**Actual Result:**
After selecting the "Select a Resource" button, the keyboard focus shifts to background elements, disrupting the navigation flow and making the interface inaccessible for keyboard users.  
**Observation**:  
The issue may be caused by the lack of a focus trap implementation for the modal or dropdown. Adding a focus trap to restrict navigation to the active area can resolve this issue. Modern browsers support the <dialog> element, which provides built-in focus trapping for modals, or you can use polyfills or custom implementations for unsupported browsers.  **Expected Result: ** 
After selecting the "Select a Resource" button, the keyboard focus should remain within the active area (e.g., modal or dropdown) until the user intentionally closes it. This can be achieved by implementing a focus trap to restrict keyboard navigation to the active area. For example:  
Use the tabindex="-1" attribute on non-focusable background elements to temporarily remove them from the focus order.  
Programmatically manage focus using JavaScript to ensure it remains within the active region.

**Relevant WCAG Criteria:** [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**
